Failing CI Test on OSX

`PythonExecutableService Ensure correct path to executable is returned`
```
AssertionError: Executable paths are not the same: expected '/usr/local/bin/python3.6' to equal '/usr/local/opt/python/bin/python3.6'
    at Object.<anonymous> (src/test/common/process/pythonProc.simple.multiroot.test.ts:124:35)
    at Generator.next (<anonymous>)
    at fulfilled (out/test/common/process/pythonProc.simple.multiroot.test.js:6:58)
    at <anonymous>
```

--- a/news/3 Code Health/1957.md
+++ b/news/3 Code Health/1957.md
@@ -1,0 +1,1 @@
+Fix failing test on Mac when validating the path of a python interperter.

--- a/src/test/common/process/pythonProc.simple.multiroot.test.ts
+++ b/src/test/common/process/pythonProc.simple.multiroot.test.ts
@@ -4,6 +4,7 @@
 import { expect, use } from 'chai';
 import * as chaiAsPromised from 'chai-as-promised';
 import { execFile } from 'child_process';
+import * as fs from 'fs-extra';
 import { Container } from 'inversify';
 import { EOL } from 'os';
 import * as path from 'path';
@@ -114,11 +115,16 @@ suite('PythonExecutableService', () => {
 
     test('Ensure correct path to executable is returned', async () => {
         const pythonPath = PythonSettings.getInstance(workspace4Path).pythonPath;
-        const expectedExecutablePath = await new Promise<string>(resolve => {
-            execFile(pythonPath, ['-c', 'import sys;print(sys.executable)'], (_error, stdout, _stdErr) => {
-                resolve(stdout.trim());
+        let expectedExecutablePath: string;
+        if (await fs.pathExists(pythonPath)) {
+            expectedExecutablePath = pythonPath;
+        } else {
+            expectedExecutablePath = await new Promise<string>(resolve => {
+                execFile(pythonPath, ['-c', 'import sys;print(sys.executable)'], (_error, stdout, _stdErr) => {
+                    resolve(stdout.trim());
+                });
             });
-        });
+        }
         const pythonExecService = await pythonExecFactory.create({ resource: workspace4PyFile });
         const executablePath = await pythonExecService.getExecutablePath();
         expect(executablePath).to.equal(expectedExecutablePath, 'Executable paths are not the same');

